instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/basic/SpringBatchTest.java

```diff
@@ -0,0 +1,283 @@
+                        .hasKind(SpanKind.INTERNAL)
+                        .hasParent(trace.getSpan(1)),
+                span ->
+                    span.hasName("BatchJob flowJob.flowStep2")
+                        .hasKind(SpanKind.INTERNAL)
+                        .hasParent(trace.getSpan(0)),
+                span ->
+                    span.hasName("BatchJob flowJob.flowStep2.Tasklet")
+                        .hasKind(SpanKind.INTERNAL)
+                        .hasParent(trace.getSpan(3))));
+  }
+
+  @Test
+  void shouldTraceSplitFlowJob() {
+    runner.runJob("splitJob");
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasName("BatchJob splitJob")
+                        .hasKind(SpanKind.INTERNAL)
+                        .hasAttribute(AttributeKey.stringKey("job.system"), "spring_batch"),
+                span ->
+                    span.satisfies(
+                            spanData ->
+                                assertThat(spanData.getName())
+                                    .matches("BatchJob splitJob.splitFlowStep[12]"))
+                        .hasKind(SpanKind.INTERNAL)
+                        .hasParent(trace.getSpan(0)),
+                span ->
+                    span.satisfies(
+                            spanData ->
+                                assertThat(spanData.getName())
+                                    .matches("BatchJob splitJob.splitFlowStep[12].Tasklet"))
+                        .hasKind(SpanKind.INTERNAL)
+                        .hasParent(trace.getSpan(1)),
+                span ->
+                    span.satisfies(
+                            spanData ->
+                                assertThat(spanData.getName())
+                                    .matches("BatchJob splitJob.splitFlowStep[12]"))
+                        .hasKind(SpanKind.INTERNAL)
+                        .hasParent(trace.getSpan(0)),
+                span ->
+                    span.satisfies(
+                            spanData ->
+                                assertThat(spanData.getName())
+                                    .matches("BatchJob splitJob.splitFlowStep[12].Tasklet"))
+                        .hasKind(SpanKind.INTERNAL)
+                        .hasParent(trace.getSpan(3))));
+  }
+
+  @Test
+  void shouldTraceJobWithDecision() {
+    runner.runJob("decisionJob");
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasName("BatchJob decisionJob")
+                        .hasKind(SpanKind.INTERNAL)
+                        .hasAttribute(AttributeKey.stringKey("job.system"), "spring_batch"),
+                span ->
+                    span.hasName("BatchJob decisionJob.decisionStepStart")
+                        .hasKind(SpanKind.INTERNAL)
```

Comment:
this 0 attribute assertion is missing for many others as well
```suggestion
                        .hasKind(SpanKind.INTERNAL)
                        .hasTotalAttributeCount(0)
```